pallets/eth-bridge/src/lib.rs

```diff
@@ -514,22 +523,30 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             ensure_none(origin)?;
 
+            ensure!(events_fraction.is_valid(), Error::<T>::InvalidEventVote);
             // Validate that the range is the current one.
             let active_range =
                 Self::active_ethereum_range().ok_or(Error::<T>::EventScanningDisabled)?;
-            ensure!(range == active_range, Error::<T>::EventScanningDisabled);
+            ensure!(range == active_range.range, Error::<T>::NonActiveEthereumRange);
+            ensure!(
+                active_range.status == EventProcessingStatus::UnderValidation,
+                Error::<T>::VotingEnded
+            );
 
             // Check if already vote has been casted
-            let votes = DiscoveredEventsShard::<T>::get(&range, &events_fraction);
+            let votes = DiscoveredEventsFractions::<T>::get(&range, &events_fraction);
+
             ensure!(votes.contains(&author.account_id) == false, Error::<T>::EventVoteExists);
-            ensure!(votes.len() as u32 >= AVN::<T>::quorum(), Error::<T>::EventVotesFull);
+            let quorum = AVN::<T>::quorum() as usize;
+            ensure!(votes.len() <= quorum, Error::<T>::EventVotesFull);
```

Comment:
We could be forgiving here and just do nothing instead of erroring?

Reply:
is `votes` the number of collators that have validated the fraction? What is the problem if we have more votes than needed?

Reply:
>We could be forgiving here and just do nothing instead of erroring?
What is the benefit of being forgiving? We save space if we just reject it.
>is votes the number of collators that have validated the fraction? What is the problem if we have more votes than needed?
yes